src/components/expression/expression-renderer.js

```diff
@@ -1338,59 +1318,121 @@ genotet.ExpressionRenderer.prototype.drawProfileLegend_ = function(profiles) {
+  this.data.profiles.forEach(function(profile, i) {
+    profile.row = i;
+  }, this);
+
+  this.data.profile.geneNames.splice(index, 1);
+  this.data.profile.values.splice(index, 1);
+  if (this.data.profile.values.length == 0) {
+    this.data.profile.valueMin = -Infinity;
+    this.data.profile.valueMax = Infinity;
+  } else {
+    var profileValues = this.data.profile.values.reduce(function(prev, next) {
+      return prev.concat(next);
+    });
+    this.data.profile.valueMin = Math.min.apply(null, profileValues);
+    this.data.profile.valueMax = Math.max.apply(null, profileValues);
+  }
+  this.drawGeneProfiles();
 };
 
 /**
  * Adds the TFA profiles for the selected genes as line charts.
  * @param {number} geneIndex
+ * @param {string} geneName
  */
-genotet.ExpressionRenderer.prototype.addTfaProfile = function(geneIndex) {
+genotet.ExpressionRenderer.prototype.addTfaProfile = function(geneIndex,
+                                                              geneName) {
   var tfaProfile = new genotet.ExpressionRenderer.Profile({
-    geneName: this.data.matrix.geneNames[geneIndex],
+    geneName: geneName,
     row: geneIndex
   });
   this.data.tfaProfiles.push(tfaProfile);
-  this.drawTfaProfiles_();
+  this.drawTfaProfiles();
 };
 
 /**
- * Adds the TFA profiles for the selected genes as line charts.
- * @param {number} geneIndex
+ * Removes the TFA profiles for the selected genes as line charts.
+ * @param {string} geneName
  */
-genotet.ExpressionRenderer.prototype.removeTfaProfile = function(geneIndex) {
+genotet.ExpressionRenderer.prototype.removeTfaProfile = function(geneName) {
   var index = -1;
   for (var i = 0; i < this.data.tfaProfiles.length; i++) {
-    if (this.data.tfaProfiles[i].row == geneIndex) {
+    if (this.data.tfaProfiles[i].geneName == geneName) {
       index = i;
       break;
     }
   }
   this.data.tfaProfiles.splice(index, 1);
-  this.drawTfaProfiles_();
+  this.data.tfaProfiles.forEach(function(tfaProfile, i) {
+    tfaProfile.row = i;
+  }, this);
+
+  this.data.tfa.geneNames.splice(index, 1);
+  this.data.tfa.tfaValues.splice(index, 1);
+  if (this.data.tfa.tfaValues.length == 0) {
+    this.data.tfa.valueMin = -Infinity;
+    this.data.tfa.valueMax = Infinity;
+  } else {
+    var profileObjects = this.data.tfa.tfaValues.reduce(function(prev, next) {
+      return prev.concat(next);
+    });
+    var profileValues = profileObjects.map(function(object) {
+      return object.value;
+    });
+    this.data.tfa.valueMin = Math.min.apply(null, profileValues);
+    this.data.tfa.valueMax = Math.max.apply(null, profileValues);
+  }
+  this.drawTfaProfiles();
+};
+
+/**
+ * Removes all the expression profiles for the selected genes as line charts.
+ */
+genotet.ExpressionRenderer.prototype.removeAllProfiles = function() {
+  this.data.profile = {
+    values: [],
+    geneNames: [],
+    conditionNames: [],
+    valueMin: Infinity,
+    valueMax: -Infinity
+  };
+  _.extend(this.data.tfa, {
```

Comment:
Any reason for re-assigning values to this.data.profile, but extending this.data.tfa? What are you trying to preserve under this.data.tfa? You should probably address in comments.